pkg/cmd/dcos_auth_listproviders.go

```diff
@@ -0,0 +1,124 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+// These are the different auth types that DC/OS supports with the names that they'll be given from the providers
+// endpoint.
+const (
+	LoginTypeDCOSUidPassword     = "dcos-uid-password"
+	LoginTypeDCOSUidServiceKey   = "dcos-uid-servicekey"
+	LoginTypeDCOSUidPasswordLDAP = "dcos-uid-password-ldap"
+	LoginTypeSAMLSpInitiated     = "saml-sp-initiated"
+	LoginTypeOIDCAuthCodeFlow    = "oidc-authorization-code-flow"
+	LoginTypeOIDCImplicitFlow    = "oidc-implicit-flow"
+)
+
+var jsonOutput bool
+
+// authListProvidersCmd represents the `dcos auth list-providers` subcommand.
+var authListProvidersCmd = &cobra.Command{
+	Use:  "list-providers",
+	Args: cobra.MaximumNArgs(1),
+	RunE: listProviders,
+}
+
+func init() {
+	authCmd.AddCommand(authListProvidersCmd)
+	authListProvidersCmd.Flags().BoolVar(&jsonOutput, "json", false, "returns providers in json format")
+}
+
+func listProviders(cmd *cobra.Command, args []string) error {
+	conf := attachedCluster().Config
```

Comment:
This is unnecessary when an argument is passed, could you execute this only when no arg is passed?